Controller/Adminhtml/Configuration/DownloadApplePayCertificate.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2022 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Controller\Adminhtml\Configuration;
+
+use Adyen\Payment\Helper\Config;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\Result\Redirect;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Filesystem\DirectoryList;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Backend\App\Action;
+use Magento\Framework\Filesystem\Io\File;
+
+class DownloadApplePayCertificate extends Action
+{
+    /**
+     * @var DirectoryList
+     */
+    private $directoryList;
+
+    /**
+     * @var File
+     */
+    private $fileIo;
+
+    /**
+     * @var Config
+     */
+    private $configHelper;
+
+    /**
+     * @param Context $context
+     * @param DirectoryList $directoryList
+     * @param Config $configHelper
+     * @param File $fileIo
+     */
+    public function __construct(
+        Context $context,
+        DirectoryList $directoryList,
+        Config $configHelper,
+        File $fileIo
+    ) {
+        parent::__construct($context);
+        $this->directoryList = $directoryList;
+        $this->configHelper = $configHelper;
+        $this->fileIo = $fileIo;
+    }
+
+    /**
+     * @param $applepayUrl
+     * @param $applepayPath
+     * @return void
+     */
+    private function downloadAndUnzip($applepayUrl, $applepayPath)
+    {
+        $tmpPath = '/tmp/xyz';
```

Comment:
We can get a dynamic temporary file with:
```suggestion
        $tmpPath = tempnam(sys_get_temp_dir(), 'apple-developer-merchantid-domain-association');
```